drivers/hmis/app/graphql/types/hmis_schema/ce_referral_step.rb

```diff
@@ -8,37 +8,72 @@
 
 module Types
   class HmisSchema::CeReferralStep < Types::BaseObject
-    # object is a Hmis::WorkflowExecution::Step
+    # object is an OpenStruct with:
+    # {
+    #   step: Hmis::WorkflowExecution::Step,
+    #   referral: Hmis::Ce::Referral,
+    # }
+    # Why: The Step schema object should resolve this step's potential Participants (ReferralParticipants on the Referral).
+    # However, the Step model does not have a direct relationship to Referral, to avoid a duplicative bidirectional relationship.
 
     field :id, ID, null: false, description: 'unique identifier for this step based on node and instance'
     field :step_id, ID, null: true, method: :id, description: 'the DB identifier of this step, if it is persisted'
     field :form_definition, Types::Forms::FormDefinition, null: false
     field :name, String, null: false
     field :status, HmisSchema::Enums::CeReferralStepStatus, null: false
-    field :swimlane, String, null: false
     field :submitted_values, JsonObject, null: true
-    delegate :name, to: :workflow_node
+    field :swimlane, HmisSchema::CeReferralSwimlane, null: true, description: 'Swimlane for this step, which holds information about potential step participants'
+    field :assignees, [Application::User], null: true, description: 'User(s) currently assigned to this step'
 
     def id
       # the step may not yet be persisted, such as when it isn't yet available in the workflow
-      "#{object.node_id}:#{object.instance_id}"
+      "#{object.step.node_id}:#{object.step.instance_id}"
+    end
+
+    def step_id
+      object.step.id
+    end
+
+    def name
+      load_ar_association(object.step, :node).name
+    end
+
+    def status
+      object.step.status
+    end
+
+    def submitted_values
+      object.step.submitted_values
     end
 
     def workflow_node
-      load_ar_association(object, :node)
+      load_ar_association(object.step, :node)
     end
 
     def swimlane
-      load_ar_association(object, :swimlane)&.name
+      swimlane = load_ar_association(object.step, :swimlane)
+      participants_by_swimlane_id = object.referral.participants.group_by(&:swimlane_id)
```

Comment:
I might be overly nit picky about this, but I find it a little confusing when a resolver method uses both `load_ar_association` AND "normal" association lookups (like `referral.participants` on the next line). I think because it's not clear to me whether this method is intended to be used in batch or not, or if there is some reason that the `swimlane` association in particular should be data loaded, while others don't need to be. (Is that the case here?)